compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -2146,13 +2146,15 @@ object Types {
   }
 
   case class LazyRef(refFn: () => Type) extends UncachedProxyType with ValueType {
+    private var myRefFn: () => Type = refFn
```

Comment:
Also, another way to avoid the leak would be to write `refFn: Context => Type` at the cost that `equals` could not be made to force the reference anymore. I tried it before and it seemed to work, but I haven't benchmarked it.